packages/types/src/index.ts

```diff
@@ -36,146 +36,579 @@ export interface Dialog {
  * Block Elements
  */
 
+/**
+ * An element to insert an image - this element can be used in section and
+ * context blocks only. If you want a block with only an image in it, you're
+ * looking for the image block.
+ *
+ * https://api.slack.com/reference/messaging/block-elements#image
+ */
 export interface ImageElement {
   type: 'image';
+
+  /**
+   * The URL of the image to be displayed.
+   */
   image_url: string;
+
+  /**
+   * A plain-text summary of the image. This should not contain any markup.
+   */
   alt_text: string;
 }
 
+/*
+ * Undocumented
+ */
 export interface UserElement {
   type: 'user';
   user_id: string;
 }
 
+/**
+ * An object containing some text
+ *
+ * https://api.slack.com/reference/messaging/composition-objects#text
+ */
 export interface PlainTextElement {
+  /**
+   * The formatting to use for this text object
+   */
   type: 'plain_text';
+
+  /**
+   * The text for the block
+   */
   text: string;
+
+  /**
+   * Indicates whether emojis in a text field should be escaped into the
+   * colon emoji format.
+   */
   emoji?: boolean;
 }
 
+/**
+ * An object containing some text
+ *
+ * https://api.slack.com/reference/messaging/composition-objects#text
+ */
 export interface MrkdwnElement {
+  /**
+   * The formatting to use for this text object
+   */
   type: 'mrkdwn';
+
+  /**
+   * The text for the block. This field accepts any of the standard
+   * [text formatting markup](https://api.slack.com/messaging/composing/formatting).
+   */
   text: string;
+
+  /**
+   * When set to false (as is default) URLs will be auto-converted into links,
+   * conversation names will be link-ified, and certain mentions will be
+   * [automatically parsed](https://api.slack.com/messaging/composing/formatting#automatic-parsing).
+   * Using a value of false will skip any preprocessing of this nature,
+   * although you can still include
+   * [manual parsing strings](https://api.slack.com/messaging/composing/formatting#advanced).
+   */
   verbatim?: boolean;
 }
 
+/**
+ * An object containing some text, formatted either as plain_text or using
+ * [Slack's "mrkdwn"](https://api.slack.com/messaging/composing/formatting).
+ *
+ * https://api.slack.com/reference/messaging/composition-objects#text
+ */
+export type TextElement = MrkdwnElement | PlainTextElement;
+
+/**
+ * An object that represents a single selectable item in a select menu.
+ *
+ * https://api.slack.com/reference/messaging/composition-objects#option
+ */
 export interface Option {
+  /**
+   * 	A plain_text only text object that defines the text shown in the option
```

Comment:
some extra space here. plus you probably don't need to specify what kind of text it is, just that it is the text displayed in the option's contents